plugins/suggested-blocks/test/suggested_blocks.mocha.js

```diff
@@ -0,0 +1,229 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+const {assert} = require('chai');
+const sinon = require('sinon');
+const Blockly = require('blockly/node');
+const SuggestedBlocks = require('../src/index');
+
+const STANDARD_TEST_CASE = {
+  usedBlockList: [
+    'controls_if',
+    'controls_if',
+    'controls_if',
+    'controls_if',
+    'controls_if',
+    'lists_length',
+    'controls_whileUntil',
+    'controls_whileUntil',
+    'controls_whileUntil',
+    'controls_whileUntil',
+    'colour_picker',
+    'colour_picker',
+    'colour_random',
+    'colour_random',
+    'colour_random',
+  ],
+  expectedFrequentBlocks: [
+    'controls_if',
+    'controls_whileUntil',
+    'colour_random',
+    'colour_picker',
+    'lists_length',
+  ],
+  expectedRecentBlocks: [
+    'colour_random',
+    'colour_picker',
+    'controls_whileUntil',
+    'lists_length',
+    'controls_if',
+  ],
+};
+
+const MANY_UNIQUE_BLOCKS_TEST_CASE = {
+  usedBlockList: [
+    'controls_if',
+    'controls_if',
+    'lists_length',
+    'controls_whileUntil',
+    'controls_whileUntil',
+    'controls_whileUntil',
+    'colour_picker',
+    'colour_picker',
+    'colour_random',
+    'colour_random',
+    'math_arithmetic',
+    'logic_operation',
+    'math_single',
+    'math_trig',
+    'math_constant',
+    'controls_repeat_ext',
+    'controls_forEach',
+    'text_multiline',
+  ],
+  expectedFrequentBlocks: [
+    'controls_whileUntil',
+    'colour_random',
+    'colour_picker',
+    'controls_if',
+    'text_multiline',
+    'controls_forEach',
+    'controls_repeat_ext',
+    'math_constant',
+    'math_trig',
+    'math_single',
+
+  ],
+  expectedRecentBlocks: [
+    'text_multiline',
+    'controls_forEach',
+    'controls_repeat_ext',
+    'math_constant',
+    'math_trig',
+    'math_single',
+    'logic_operation',
+    'math_arithmetic',
+    'colour_random',
+    'colour_picker',
+  ],
+};
+
+const FREQUENCY_TIEBREAK_TEST_CASE = {
+  usedBlockList: [
+    // Used once
+    'controls_if',
+    'lists_length',
+    'controls_whileUntil',
+    'colour_picker',
+    'colour_random',
+    'math_arithmetic',
+    // Used a second time
+    'controls_if',
+    'controls_whileUntil',
+    'colour_picker',
+    'colour_random',
+    // Used a third time
+    'colour_picker',
+    'colour_random',
```

Comment:
The third comment has confused me. Is `colour_picker` meant to be `controls_if`?

Reply:
the ones that are used a third time are a subset of the controls in the "used once" list. They're not necessarily the first ones in the previous lists, to add more fuzz to the testing.

Reply:
Ohhh I see what's happening. Thanks for clarifying!